src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,498 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-            if (volume && volume.format) {
-                stateDelta.format = volume.format;
-                if (volume.format === "iso")
-                    deviceType = "cdrom";
+        if (!newBus) {
+            // If disk with the same device exists, use the same bus too
+            for (const disk of Object.values(vm.disks)) {
+                if (disk.device === diskParams.device) {
+                    newBus = disk.bus;
+                    break;
+                }
             }
+            newBus = newBus || "virtio";
         }
 
-        this.onValueChanged("device", deviceType);
+        setDiskParams(diskParams => ({ ...diskParams, busType: newBus }));
+    }, [diskParams.device, vm.disks]);
 
-        return stateDelta;
-    }
+    useEffect(() => {
+        // Initial state settings and follow up state updates after 'busType' is changed
+        const existingTargets = Object.getOwnPropertyNames(vm.disks);
+        const availableTarget = isMediaInsertion ? disk.target : getNextAvailableTarget(existingTargets, diskParams.busType);
+        setDiskParams(diskParams => ({ ...diskParams, target: availableTarget }));
+    }, [vm.disks, disk?.target, diskParams.busType, isMediaInsertion]);
 
-    onValueChanged(key, value) {
-        let stateDelta = {};
-        const { storagePools, vm } = this.state;
+    const currentPoolRef = useRef();
+    useEffect(() => {
+        if (currentPoolRef.current === undefined) {
+            currentPoolRef.current = diskParams.storagePool;
 
-        switch (key) {
-        case 'storagePoolName': {
-            const currentPool = storagePools.find(pool => pool.name === value && pool.connectionName === vm.connectionName);
-            const prevPool = storagePools.find(pool => pool.name === this.state.storagePoolName && pool.connectionName === vm.connectionName);
-            this.setState({ storagePoolName: value, storagePoolType: currentPool.type });
             // Reset the format only when the Format selection dropdown changes entries - otherwise just keep the old selection
             // All pool types apart from 'disk' have either 'raw' or 'qcow2' format
-            if (currentPool && prevPool && ((currentPool.type == 'disk' && prevPool.type != 'disk') || (currentPool.type != 'disk' && prevPool.type == 'disk'))) {
-                // use onValueChange instead of setState in order to perform subsequent state change logic
-                this.onValueChanged('format', getDefaultVolumeFormat(value));
+            const prevPool = currentPoolRef.current;
+            if ((diskParams.storagePool?.type == 'disk' && prevPool?.type != 'disk') || (diskParams.storagePool?.type != 'disk' && prevPool?.type == 'disk')) {
+                prevPool.current = diskParams.storagePool;
+                setDiskParams(diskParams => ({
+                    ...diskParams,
+                    format: getDefaultVolumeFormat(diskParams.storagePool?.name),
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231117-134447-13cb53e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)